samples/KvsAacAudioVideoStreamingSample.c

```diff
@@ -0,0 +1,300 @@
+    streamStopTime = defaultGetTime() + streamingDuration;
+
+    // default storage size is 128MB. Use setDeviceInfoStorageSize after create to change storage size.
+    CHK_STATUS(createDefaultDeviceInfo(&pDeviceInfo));
+    // adjust members of pDeviceInfo here if needed
+    pDeviceInfo->clientInfo.loggerLogLevel = LOG_LEVEL_DEBUG;
+
+    CHK_STATUS(createRealtimeAudioVideoStreamInfoProvider(streamName, DEFAULT_RETENTION_PERIOD, DEFAULT_BUFFER_DURATION, &pStreamInfo));
+
+    // adjust members of pStreamInfo here if needed
+
+    // set up audio cpd.
+    for(UINT32 i = 0; i < 2; ++i) {
+        if (pStreamInfo->streamCaps.trackInfoList[i].trackId == DEFAULT_AUDIO_TRACK_ID) {
+            pStreamInfo->streamCaps.trackInfoList[i].codecPrivateData = audioCpd;
+            pStreamInfo->streamCaps.trackInfoList[i].codecPrivateDataSize = ARRAY_SIZE(audioCpd);
+        }
+    }
+    // use relative time mode. Buffer timestamps start from 0
+    pStreamInfo->streamCaps.absoluteFragmentTimes = FALSE;
+
+    CHK_STATUS(createDefaultCallbacksProviderWithAwsCredentials(accessKey,
+                                                                secretKey,
+                                                                sessionToken,
+                                                                MAX_UINT64,
+                                                                region,
+                                                                cacertPath,
+                                                                NULL,
+                                                                NULL,
+                                                                FALSE,
+                                                                &pClientCallbacks));
+
+    CHK_STATUS(createKinesisVideoClient(pDeviceInfo, pClientCallbacks, &clientHandle));
+    CHK_STATUS(createKinesisVideoStreamSync(clientHandle, pStreamInfo, &streamHandle));
+
+    data.streamStopTime = streamStopTime;
+    data.streamHandle = streamHandle;
+    data.streamStartTime = defaultGetTime();
+    ATOMIC_STORE_BOOL(&data.firstVideoFramePut, FALSE);
+
+    THREAD_CREATE(&videoSendTid, putVideoFrameRoutine,
+                          (PVOID) &data);
+    THREAD_CREATE(&audioSendTid, putAudioFrameRoutine,
+                          (PVOID) &data);
+
+    THREAD_JOIN(videoSendTid, NULL);
+    THREAD_JOIN(audioSendTid, NULL);
+
+    CHK_STATUS(stopKinesisVideoStreamSync(streamHandle));
+    CHK_STATUS(freeKinesisVideoStream(&streamHandle));
+    CHK_STATUS(freeKinesisVideoClient(&clientHandle));
+
+CleanUp:
+
+    if (STATUS_FAILED(retStatus)) {
+        defaultLogPrint(LOG_LEVEL_ERROR, "", "Failed with status 0x%08x\n", retStatus);
+    }
+
+    for(i = 0; i < NUMBER_OF_AAC_FRAME_FILES; ++i) {
+        SAFE_MEMFREE(data.audioFrames[i].buffer);
+    }
+
+    for(i = 0; i < NUMBER_OF_H264_FRAME_FILES; ++i) {
+        SAFE_MEMFREE(data.videoFrames[i].buffer);
+    }
+
+    if (pDeviceInfo != NULL) {
```

Comment:
For all of these cases, just do freeXXX - those are idempotent. Just make sure the pBlah is initialized to NULL first